ext/distributed_tracing_headers.c

```diff
@@ -42,6 +43,54 @@ static void dd_check_tid(ddtrace_distributed_tracing_result *result) {
     }
 }
 
+static void ddtrace_deserialize_baggage(char *baggage_ptr, char *baggage_end, HashTable *baggage) {
+    while (baggage_ptr < baggage_end) {
+        char *key_start = baggage_ptr;
+        while (baggage_ptr < baggage_end && *baggage_ptr != '=') {
+            ++baggage_ptr;
+        }
+        if (baggage_ptr >= baggage_end || *baggage_ptr != '=') break;
```

Comment:
```suggestion
        if (baggage_ptr >= baggage_end) break;
```
This will never be `!= '='` at this place